pkg/apiserver/audit/audit.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audit
+
+import (
+	"bufio"
+	"net"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+
+	"k8s.io/kubernetes/pkg/api"
+	utilnet "k8s.io/kubernetes/pkg/util/net"
+)
+
+// auditResponseWriter implements http.ResponseWriter interface.
+type auditResponseWriter struct {
+	http.ResponseWriter
+	id string
+}
+
+func (a *auditResponseWriter) WriteHeader(code int) {
+	glog.Infof("AUDIT: id=%q response=\"%d\"", a.id, code)
```

Comment:
Just don't use glog may be the easiest :)

On Thu, Jun 9, 2016 at 7:36 AM, Maciej Szulik <notifications@github.com>
wrote:

> In pkg/apiserver/audit/audit.go
> <https://github.com/kubernetes/kubernetes/pull/27087#discussion_r66452677>
> :
>
> > +
> > +	"github.com/golang/glog"
> > +	"github.com/pborman/uuid"
> > +
> > +	"k8s.io/kubernetes/pkg/api"
> > +	utilnet "k8s.io/kubernetes/pkg/util/net"
> > +)
> > +
> > +// auditResponseWriter implements http.ResponseWriter interface.
> > +type auditResponseWriter struct {
> > +	http.ResponseWriter
> > +	id string
> > +}
> > +
> > +func (a *auditResponseWriter) WriteHeader(code int) {
> > +	glog.Infof("AUDIT: id=%q response=\"%d\"", a.id, code)
>
> The log is already hard enough to read-- I think I'd prefer a separate
> file.
>
> I don't mind that, can you please tell me how to do it with glog though? I
> haven't see any possibility there :( But I agree with Clayton about placing
> it behind an interface.
>
> Also I think it might be good to put the audit ID in the context? I guess
> we can do that when we have a use case.
>
> Makes sense as well.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27087/files/752c551c39be25f16e03d37bec8a3c2567467370#r66452677>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAnglhR5eD-2XOT8-YcGg8_0JQ1dW5Myks5qKCTVgaJpZM4Ixcs->
> .
>